src/profile-logic/process-profile.js

```diff
@@ -771,12 +771,23 @@ function _processMarkerPayload(
  * Explicitly recreate the markers here to help enforce our assumptions about types.
  */
 function _processSamples(geckoSamples: GeckoSampleStruct): SamplesTable {
-  return {
-    responsiveness: geckoSamples.responsiveness,
+  const samples: SamplesTable = {
     stack: geckoSamples.stack,
     time: geckoSamples.time,
     length: geckoSamples.length,
   };
+
+  if (geckoSamples.eventDelay) {
+    samples.eventDelay = geckoSamples.eventDelay;
+  } else if (geckoSamples.responsiveness) {
+    samples.responsiveness = geckoSamples.responsiveness;
+  } else {
+    throw new Error(
+      'There is no eventDelay or responsiveness array in the samples table!'
```

Comment:
I might change the wording on this as more of an expectation. This way future developers will feel free to change the implementation if they need to, and understand the context a bit more. "The profile processor expected an eventDelay or responsiveness array in the samples table, but none was found."

I'm happy to keep this throw here.